source/NVDAObjects/IAccessible/__init__.py

```diff
@@ -1560,11 +1560,24 @@ def _get_isPresentableFocusAncestor(self):
 class TrayClockWClass(IAccessible):
 	"""
 	Based on NVDAObject but the role is changed to clock.
+	The value contains the time, and the date if available.
 	"""
 
 	def _get_role(self):
 		return controlTypes.ROLE_CLOCK
 
+	def _get_name(self):
+		#The name contains redundant information that is available either in the role or the value.
+		return None
+
+	def _get_value(self):
+		value = super(TrayClockWClass, self).value
+		if value is None:
+			#On XP there is no value, fall back to using the name.
+			return super(TrayClockWClass, self).name
+		#Strip Unicode left-to-right and right-to-left marks.

```

Comment:
Good point, will add that.

 

From: josephsl [mailto:notifications@github.com] 
Sent: Sunday, April 3, 2016 10:12
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Davy Kager <mail@davykager.nl>
Subject: Re: [nvaccess/nvda] Improve reading of the clock in the taskbar (#5751)

 

In source/NVDAObjects/IAccessible/__init__.py <https://github.com/nvaccess/nvda/pull/5751#discussion_r58306141> :

>       """
>  
>       def _get_role(self):
>              return controlTypes.ROLE_CLOCK
>  
> +     def _get_name(self):
> +            #The name contains redundant information that is available either in the role or the value.
> +            return None
> +
> +     def _get_value(self):
> +            value = super(TrayClockWClass, self).value
> +            if value is None:
> +                    #On XP there is no value, fall back to using the name.
> +                    return super(TrayClockWClass, self).name
> +            #Strip Unicode left-to-right and right-to-left marks.

I think referencing a ticket might help future devs to find out why the below code is necessary.

—
You are receiving this because you authored the thread.
Reply to this email directly or  <https://github.com/nvaccess/nvda/pull/5751/files/19b600d36da6a6f2557663dc605a251fad2086c6#r58306141> view it on GitHub  <https://github.com/notifications/beacon/AC9y8fJ8BqF7uVD9EksepuqFARlO3zU-ks5pz3ZAgaJpZM4HZsGx.gif>